libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2230,23 +2230,24 @@ bool TypeChecker::visit(FunctionCall const& _functionCall)
 		}
 		}
 
-		returnType = returnTypes.size() == 1 ?
+		funcCallAnno.type = returnTypes.size() == 1 ?
 			move(returnTypes.front()) :
 			make_shared<TupleType>(move(returnTypes));
+
+		break;
 	}
-	else
-	{
+
+	default:
 		// REVIEW - original code set function call kind to function call for non-callables
-		_functionCall.annotation().kind = FunctionCallKind::FunctionCall;
-	}
+		funcCallAnno.kind = FunctionCallKind::FunctionCall;
```

Comment:
I agree, it is probably better to set this to `FunctionCall`, but in this case, please add a `solAssert(m_errorReporter.hasError(), "")` to ensure that we already reported an error earlier.

Reply:
Thanks, I have committed that change.